warrior/Framework/ClassUtils/configuration_element_class.py

```diff
@@ -73,7 +76,8 @@ def __expand_variables(self, string):
 
             if match is None:
                 clean_partial += return_value[:return_value.find(check) + len(check)]
-                return_value = return_value.replace(return_value[:return_value.find(check) + len(check)], '')
+                # replace only the clean_partial value when there are multiple '}'
+                return_value = return_value.replace(return_value[:return_value.find(check) + len(check)], '', 1)
                 match = self.__find_match(return_value[:return_value.find(check) + len(check)])
```

Comment:
This whole block (if match is None) looks unnecessary code. Can you please check why this is necessary? If not necessary remove and that would solve the original issue as well.

Reply:
It is necessary in cases like, if we have "ab {abc} cd ${list.value}". So we cant remove those lines.